File: src/etch/bindings/java/support/TypeMap.java
package etch.bindings.java.support;

import etch.bindings.java.msg.Type;

/**
 * TypeMap is a IdNameMap for Type.
 */
public final class TypeMap extends IdNameMap<Type>
{
	@Override
	public Type makeNew( String name )
	{
		return new Type( name );
	}
}

File: src/etch/bindings/csharp/support/TypeMap.cs
using System;
using System.Collections.Generic;
using System.Text;
using Etch.Msg;

namespace Etch.Support
{
    /// <summary>
    /// TypeMap is an IdNameMap for Type
    /// </summary>
    public class TypeMap : IdNameMap<XType>
    {
        public override XType MakeNew( string name )
        {
            return new XType( name );
        }
    }
}
